### PR TITLE
[mBART] skip broken forward pass test, stronger integration test

### PR DESCRIPTION
**Pull request by sshleifer**
(1) I fixed the config of mbart to go from BLEU score 12 to BLEU score 26. This requires that the `decoder_start_token_id=lang_code['ro']` but that we filter that out when we decode.
(2) I added another, harder, integration test case. The expected result is only 1 word different than GT. This also adds implicit coverage for batched generation.
(3) tried for awhile to fix the broken slow GPU `test_enro_forward`. 
On commits where the test passed, it doesn't pass any more. And the model hadn't been updated on S3 since March 26. So for now we skip it.

### CODE REVIEW

File: src/transformers/tokenization_bart.py [hunk 1]
```diff
@@ -110,6 +110,12 @@ class MBartTokenizer(XLMRobertaTokenizer):
     id_to_lang_code = {v: k for k, v in lang_code_to_id.items()}
     cur_lang_code = lang_code_to_id["en_XX"]
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fairseq_tokens_to_ids.update(self.lang_code_to_id)
+        self.fairseq_ids_to_tokens = {v: k for k, v in self.fairseq_tokens_to_ids.items()}
+        self._additional_special_tokens = list(self.lang_code_to_id.keys())
```
Comment (sshleifer) [inline src/transformers/tokenization_bart.py:117]: allows language codes to be filtered out by `tokenizer.decode(skip_special_tokens=True)`

File: src/transformers/tokenization_bart.py [hunk 2]
```diff
@@ -118,12 +124,6 @@ def build_inputs_with_special_tokens(self, token_ids_0, token_ids_1=None) -> Lis
         # We don't expect to process pairs, but leave the pair logic for API consistency
         return token_ids_0 + token_ids_1 + special_tokens
 
-    def _convert_id_to_token(self, index):
```
Comment (sshleifer) [inline src/transformers/tokenization_bart.py:121]: now same as parent implem

File: src/transformers/tokenization_bart.py [hunk 3]
```diff
@@ -159,6 +159,7 @@ def prepare_translation_batch(
             return_tensors=return_tensors,
             max_length=max_length,
             pad_to_max_length=pad_to_max_length,
+            truncation=True,
```
Comment (sshleifer) [inline src/transformers/tokenization_bart.py:162]: avoid warning

File: tests/test_modeling_bart.py [hunk 4]
```diff
@@ -247,22 +251,22 @@ def test_enro_forward(self):
...
         net_input["attention_mask"] = net_input["input_ids"].ne(self.pad_token_id)
         with torch.no_grad():
             logits, *other_stuff = model(**net_input)
 
-        expected_slice = [9.0078, 10.1113, 14.4787]
-        result_slice = logits[0][0][:3].tolist()
-        self.assertListEqual(expected_slice, result_slice)
+        expected_slice = torch.tensor([9.0078, 10.1113, 14.4787], device=logits.device, dtype=logits.dtype)
+        result_slice = logits[0, 0, :3]
+        _assert_tensors_equal(expected_slice, result_slice, atol=TOLERANCE)
```
Comment (sshleifer) [inline tests/test_modeling_bart.py:261]: addresses @patrickvonplaten comment from old PR

**Comment by patrickvonplaten** [comment]
Looks good to me

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**